freqtrade/freqtradebot.py

```diff
@@ -473,6 +473,14 @@ def check_and_call_adjust_trade_position(self, trade: Trade):
         If the strategy triggers the adjustment, a new order gets issued.
         Once that completes, the existing trade is modified to match new data.
         """
+        if self.strategy.max_buy_position_adjustment > -1:
+            filled_buys = trade.select_filled_orders('buy')
+            count_of_buys = len(filled_buys)
+            if count_of_buys > self.strategy.max_buy_position_adjustment:
```

Comment:
Are you sure this comparison is correct?

assuming we set `max_buy_position_adjustment` to 3 (so we allow 3 buy orders - for example 3x 100$ for a total of 300$ stake)
after the initial trade entry, `count_of_buys` will be 1
after the 2nd order, it'll be 2
after the 3rd order, i'll be 3 (we've now reached our 3x 100$ investment)

This comparison will however only result to true on the 4th order - basically having the investment as 4x 100$.

Now it'll depend if we do count the initial buy or not. 
Personally, i think it feels more intuitive to define the overall entries (including the initial call) ... otherwise i have to calculate my total stake per pair as `(1 + max_buy_position_adjust) * stake` (assuming an even stake, obviously).

Reply:
It depends on the name of the setting, in my opinion. max_buy_position_adjustment for me refers to the extra buys that can only happen if we enable position adjustment, hence the naming.

If we want to go for total number of buys, then naming it max_buys_per_trade would be better?

Reply:
that's correct - i guess both are valid, we just need to be clear in the documentation.
i've also asked for additional opinions in discord ...

Reply:
This could be written a lot simpler like this, because we have nr_of_successful_buys:
`if trade.nr_of_successful_buys > self.strategy.max_buy_position_adjustment:`

Also, since our trade already contains "nr_of_successful_buys" then it would make sense to name the max variable as "max_nr_of_trade_buys" or "max_nr_of_buys_per_trade"

Reply:
that's "danger zone" i think

calling it `max_buys_per_trade` will have a different impact than `max_buy_position_adjustment` ... as i'd expect the first to include the initial order, while the 2nd wouldn't.

Reply:
I'd be ok with both names. 
Also wanted to point out that we also have /forcebuy that allows buying additional orders and that would not (and should not) honor this max limit, since it's an end-user initiated Force buy (I always imagine the user doing a Jedi mind-trick pose when using that).

Reply:
The name is now Fine based on the description provided.

This was more about making sure this is what we want.